activerecord/test/cases/base_test.rb

```diff
@@ -1352,6 +1352,29 @@ def test_slice_accepts_array_argument
     assert_equal attrs, topic.slice(attrs.keys)
   end
 
+  def test_values_at
+    attrs = {
+      title: "Budget",
+      author_name: "Jason"
+    }.with_indifferent_access
+
+    topic = Topic.new(attrs)
+
+    assert_equal(%w( Budget Jason ), topic.values_at(:title, :author_name))
+    assert_equal(%w( Jason Budget ), topic.values_at(:author_name, :title))
```

Comment:
We need test coverage for the behaviour of calling methods, rather than returning attribute values.

```suggestion
    assert_equal([ "Budget", "Jason", "I am Jack's profound disappointment" ], topic.values_at(:title, :author_name, :arbitrary_method))
    assert_equal([ "I am Jack's profound disappointment", "Jason", "Budget" ], topic.values_at(:arbitrary_method, :author_name, :title))
```

Reply:
Hum why would I do that? It would raise an `NoMethodError` on `arbitrary_method`

Reply:
Oh, I missed that these tests use a different model than the ones for `slice` above.

We need to test that `values_at` calls methods, rather than extracting values from the attributes hash directly, in the same way that `slice` does. It looks like there's an existing `topic_id` method that we could use for this:

https://github.com/rails/rails/blob/badba9d2fa90ae96a5d8428b0a739e18b1581607/activerecord/test/models/topic.rb#L63-L65